core/cmd/local_client.go

```diff
@@ -557,7 +544,14 @@ func (cli *Client) RebroadcastTransactions(c *clipkg.Context) (err error) {
 		return err
 	}
 
-	err = keyStore.Unlock(pwd)
+	if c.IsSet("password") {
+		pwd, err := utils.PasswordFromFile(c.String("password"))
+		if err != nil {
+			return cli.errorOut(fmt.Errorf("error reading password: %+v", err))
+		}
+		cli.Config.SetPasswords(&pwd, nil)
+	}
+	err = keyStore.Unlock(cli.Config.KeystorePassword())
```

Comment:
Should this command do its own validation via `cli.Config.Validate()` since that is not done in the before anymore?
Do the DB commands need to do the same?

Reply:
Oh, actually validation was never done in the root level before? So it has always been missing here. Probably never came up since this is normally run alongside an already working deployment and just reuses it.

Reply:
correct. i actually tried adding Validate before requesting review and it broke existing tests.

Reply:
Oh, how did it break? I think it is pretty important to have here since this command uses some complex systems that depend on invariants enforced by validation.

Reply:
```   	Error Trace:	/Users/kreherma/git/cll/chainlink/core/cmd/local_client_test.go:348
        	Error:      	Received unexpected error:
        	            	error validating configuration: 2 errors:
        	            		- invalid environment: environment variable DATABASE_URL must not be set: unsupported with config v2
        	            		- invalid configuration: EVM.0.Nodes.0: 3 errors:
        	            						- Name: missing: required for all nodes
        	            						- WSURL: missing: required for primary nodes
        	            						- HTTPURL: missing: required for all nodes
        	Test:       	TestClient_RebroadcastTransactions_OutsideRange_Txm/below_beginning```
        	
        	it wasn't immediately clear were the config is being created in that test or i would have simply fixed it. i was expecially leary of the DATABASE_URL error because it's still explicitly referenced and set in various tests.
        	
        	maybe we should have another ticket to clean that up?

Reply:
Ah this test mocks a bunch of stuff... in that case maybe we can just pass meaningless test values to satisfy the validation?

Reply:
i was mistaken. i thought DATABASE_URL was being set in the bowels of our test harness, which i was afraid to touch in this PR. Instead, i had it set in my env. 

Config.Validate added and test updated

i think there is still another ticket or two for other cleanup including
- db cmds
- password handling/config mutation (we have cp & pasted code in at least 2 cmds, and i'd like all the config mutation to be in the Before func)
- comb'ing the code for DATABASE_URL and updating to CL_DATABASE_URL

Reply:
SGTM
Yeah I was hoping we could use the before, but do the client commands get in the way? I think it would have to know which subcommand is running to skip requiring anything them - sounds messy but maybe doable :shrug:.